### PR TITLE
Link docs on parsing failure (#886)

### PR DESCRIPTION
**Pull request by beckjake**
When we fail to parse the schema.yml file for reasons that are probably version-related, raise an error that includes a link to the docs.

I also added some checks to raise `dbt.exceptions.CompilationError` instead of `AttributeError` that I found while I was testing/implementing, because they were incredibly unhelpful errors as-is.

### CODE REVIEW

File: dbt/parser/schemas.py [hunk 1]
```diff
@@ -343,6 +361,14 @@ def parse_v2_yml(cls, original_file_path, test_yml, package_name,
             return
 
         for model in test_yml['models']:
+            if not isinstance(model, dict):
+                msg = (
+                    "Invalid test config given in {} near {} (expected a dict)"
+                ).format(original_file_path, model)
+                if dbt.flags.STRICT_MODE:
+                    dbt.exceptions.raise_compiler_error(msg, model)
+                dbt.utils.compiler_warning(model_name, msg)
```
[the remark this answers is not in the document]
Comment (beckjake) [reply dbt/parser/schemas.py]: oops, fixed

**Comment by beckjake** [comment]
Getting `UnparsedNodeUpdate` out of the error messages will require a bit more work - we'll have to catch and re-raise validation errors and (I think) parse their messages.

**Comment by drewbanin** [comment]
hmm, I think it would be great to clean up this error message if possible but I don't think it's worth sinking a ton of time into.

My big qualm is with potentially the full test definition showing up in the `Invalid test config given...` line. Can we just change this warning to not include the `near {configs}` in stdout? https://github.com/fishtown-analytics/dbt/blob/development/dbt/parser/schemas.py#L301

Additionally, if there's a way to log the filename where the file occurred, then I think this will be a really helpful and actionable error message. You buy all that?

**Comment by beckjake** [comment]
Yeah, I think so.

File: dbt/parser/schemas.py [hunk 2]
```diff
@@ -343,13 +361,21 @@ def parse_v2_yml(cls, original_file_path, test_yml, package_name,
...
             try:
                 model = UnparsedNodeUpdate(**model)
-            except dbt.exceptions.ValidationException as exc:
+            except dbt.exceptions.JSONValidationException as exc:
                 # we don't want to fail the full run, but we do want to fail
                 # parsing this file
-                msg = "Invalid test config given in {} near {}: {}".format(
-                        original_file_path, model, exc
+                msg = "Invalid test config given in {}: {}".format(
+                        original_file_path, exc.errors_message
```
Comment (drewbanin) [inline dbt/parser/schemas.py]: I think the warning we show two lines below here is kind of confusing. Presently, dbt shows the following warning in this code path:
```
* Compilation warning while compiling model event:
* Invalid test config given in models/schema.yml: Additional properties are not allowed ('descriptin' was unexpected)
```

This is a test, not a mode, so I can imagine someone being confused when they see this warning. Can we just quickly change the `compiler_warning` in `utils.py` to accept a resource type?

Then we could do:
```
dbt.utils.compiler_warning(model.get('name', 'unknown'), msg, resource_type='test')
```

File: dbt/exceptions.py [hunk 3]
```diff
@@ -421,3 +435,15 @@ def raise_duplicate_patch_name(name, patch_1, patch_2):
...
+
+
+def raise_incorrect_version(path):
+    raise_compiler_error(
+        'The schema file at {} does not contain a version specifier. dbt '
+        'assumes that schema.yml files without version specifiers are version '
+        '1 schemas, but this file looks like a version 2 schema. If this is '
+        'the case, you can fix this error by adding `version: 2` to the top '
+        'of the file.\n\nOtherwise, please consult the documentation for more '
+        'information on schema.yml syntax: '
```
Comment (drewbanin) [inline dbt/exceptions.py]: can we just stick two newlines between the end of the sentence and the next line? It makes it easier to copy/paste

**Comment by drewbanin** [comment]
@beckjake two more very tiny comments and then this looks good to go! I'm going to approve it -- ship when ready!

**Pull request merged**